Simplenote/NSMutableAttributedString+Styling.m

```diff
@@ -21,7 +25,11 @@ - (void)processChecklistsWithColor:(UIColor *)color
                                           range:plainString.fullRange] reverseObjectEnumerator] allObjects];
 
     for (NSTextCheckingResult *match in matches) {
-        NSRange matchedRange = match.range;
+        if (ListRangeIndex >= match.numberOfRanges) {
```

Comment:
Currently, we expect two matches and the last match is the list.  Under these circumstances comparing ListRangeIndex with the numberOfRanges in this way is sound. I'm wondering how it could be future proof.  Consider that at some point in the future more than two ranges are expected (who knows why) and maybe in some future scenario the ListRangeIndex is not 1, and is not the last match. 
My concern is we have a const with a specific purpose but we're using it for two different things. 
I'm fine if we want to keep it like this, but maybe we could add a comment about how its being used here for the benefit of future engineers?

Reply:
I'll add another constant with the expected number of ranges, just for future proof purposes, and will also add a comment detailing what's going on there. Thank you sir!!!